test/WebJobs.Script.Tests.Integration/Host/SecretsRepositoryTests.cs

```diff
@@ -370,6 +369,133 @@ public async Task GetSecretSnapshots_ReturnsExpected(SecretsRepositoryType repos
+                Assert.Equal(secretsContent.MasterKey.Value, "test");
+                Assert.Equal(secretsContent.FunctionKeys[0].Name, KeyName);
+                Assert.Equal(secretsContent.FunctionKeys[0].Value, "test");
+                Assert.Equal(secretsContent.SystemKeys[0].Name, KeyName);
+                Assert.Equal(secretsContent.SystemKeys[0].Value, "test");
+            }
+        }
+
+        [Theory]
+        [InlineData(SecretsRepositoryType.BlobStorage)]
+        [InlineData(SecretsRepositoryType.BlobStorageSas)]
+        public async Task BlobRepository_SimultaneousWrites_Throws_PreconditionFailed(SecretsRepositoryType repositoryType)
+        {
+            using (var directory = new TempDirectory())
+            {
+                await _fixture.TestInitialize(repositoryType, directory.Path);
+
+                HostSecrets testSecrets = new HostSecrets()
+                {
+                    MasterKey = new Key("master", "test"),
+                    FunctionKeys = new List<Key>() { new Key(KeyName, "test") },
+                    SystemKeys = new List<Key>() { new Key(KeyName, "test") }
+                };
+
+                string testFunctionName = "host";
+
+                var target = _fixture.GetNewSecretRepository();
+
+                // Set up initial secrets.
+                await _fixture.WriteSecret(testFunctionName, testSecrets);
+                HostSecrets secretsContent = await target.ReadAsync(ScriptSecretsType.Host, testFunctionName) as HostSecrets;
+                Assert.Equal("test", secretsContent.FunctionKeys.Single().Value);
+
+                testSecrets.FunctionKeys.Single().Value = "changed";
+
+                // Simultaneous writes will result in one of the writes being discarded due to
+                // non-matching ETag.
+                Task writeTask1 = target.WriteAsync(ScriptSecretsType.Host, testFunctionName, testSecrets);
+                Task writeTask2 = target.WriteAsync(ScriptSecretsType.Host, testFunctionName, testSecrets);
+
+                var ex = await Assert.ThrowsAsync<RequestFailedException>(() => Task.WhenAll(writeTask1, writeTask2));
+
+                // Ensure the write went through.
+                secretsContent = await target.ReadAsync(ScriptSecretsType.Host, testFunctionName) as HostSecrets;
+                Assert.Equal("changed", secretsContent.FunctionKeys.Single().Value);
+
+                Assert.Equal("ConditionNotMet", ex.ErrorCode);
+                Assert.Equal(412, ex.Status);
+                Assert.True(writeTask1.IsCompletedSuccessfully || writeTask2.IsCompletedSuccessfully,
+                    "One of the write operations should have completed successfully.");
+            }
+        }
+
+        [Theory]
+        [InlineData(SecretsRepositoryType.BlobStorage)]
+        [InlineData(SecretsRepositoryType.BlobStorageSas)]
+        public async Task BlobRepository_SimultaneousCreates_Throws_Conflict(SecretsRepositoryType repositoryType)
```

Comment:
With the previous code in `WriteToBlobAsync()`, this test behaved slightly different -- rather than throw a 409, it threw a 412. I don't see this as a problem, just wanted to call it out.